src/fmu/dataio/dataio.py

```diff
@@ -763,11 +765,16 @@ def _establish_pwd_rootpath(self) -> None:
             logger.info("The casepath is hard set as %s", self._rootpath)
 
         else:
-            if self._inside_rms or INSIDE_RMS or "RUN_DATAIO_EXAMPLES" in os.environ:
+            if ExportData._inside_rms or INSIDE_RMS or DATAIO_EXAMPLES:
+                logger.info(
+                    "Run from inside RMS: ExportData._inside_rms=%s, "
+                    "INSIDE_RMS=%s, DATAIO_EXAMPLES=%s",
+                    ExportData._inside_rms,
+                    INSIDE_RMS,
+                    DATAIO_EXAMPLES,
+                )
                 self._rootpath = (self._pwd / "../../.").absolute().resolve()
-                logger.info("Run from inside RMS (or pretend)")
-                # BUG(?): Should be ExportData._inside_rms ?
-                self._inside_rms = True  # type: ignore[misc]
+                ExportData._inside_rms = True
```

Comment:
Consider the below ex.

```python
from dataclasses import dataclass
from typing import ClassVar


@dataclass
class Bar:
    x: ClassVar[str] = "x-unset"
    y: str = "y-unset"


b = Bar()
print(b, b.x)
b.x = "x-set"
print(Bar.x)
Bar.x = "x-set"
print(Bar.x)
>>>
Bar(y='y-unset') x-unset
x-unset
x-set

```

If we update the instance reference of x, it will only be updated for that instance of ExportData.

Reply:
I cannot see any situation where we would like to set this on one instance of ExportData only in the context of the same script. That said, I believe this is for testing purposes only, to emulate fmu-dataio being ran inside the RMS environment. Fixing this sounds like a rational thing to do, but this also feels like an issue more than a PR.

Reply:
JB 👍 . Yes these are used for testing only, but correct syntax is important still.